File: solutions/cpp/series/1/series.cpp
#include "series.h"
#include <stdexcept>
#include <cmath>

using namespace std;

namespace series {

    vector<string>slice (string input, int length) {
        unsigned int len = abs(length);
        if(input.empty()) { 
            throw(domain_error("Empty series is invalid."));
        } else if (length<=0) {
            throw(domain_error("Slice length must be positive."));
        } else if(input.length() < len) {
            throw(domain_error("Slice length too large."));
        }
        vector<string> result;
        for(unsigned int i=0; i<input.length()-len+1; i++){
            result.emplace_back(input.substr(i, len));
        }
        return result;
    }
 
}  // namespace series

File: solutions/kotlin/series/1/src/main/kotlin/Series.kt
object Series {

    fun slices(n: Int, s: String): List<List<Int>> {
        if (s=="")
            throw IllegalArgumentException("Empty series is invalid.")
        if (n<=0)
            throw IllegalArgumentException("Slice length must be positive.")
        if (s.length < n)
            throw IllegalArgumentException("Slice length is too large.")
        var result = mutableListOf<List<Int>>()
        for(i in 0..(s.length-n)) {
            result.add(s.substring(i,i+n).map{ch -> ch.digitToInt()})
        }
        return result
    }
}
